File: heap/estatico/Java/Árvore Binária Heap Genérico Máximo/Amontoavel.java
//Heap = Monte
//Heapify = Organizar dados na forma de um Heap
//Heapifiable = Amontoavel
public interface Amontoavel<T> {
	//insert
	void inserir(T dado);
	//extract
	T extrair();
	//get
	T obterRaiz();
	
	//print
	String imprimir();
	//isEmpty
	boolean estaVazia();
	//isFull
	boolean estaCheia();
}

File: heap/estatico/Kotlin/Arvore Binária Heap Máximo Paciente/Amontoavel.kt
public interface Amontoavel {
	fun inserir(dado: Int) 	//insert	C
	fun extrair(): Int?		//extract	D
	fun obter(): Int?		//get		R
	fun atualizar(dado: Int)//update	U
	
	fun imprimir(): String
	fun estaVazia(): Boolean
	fun estaCheia(): Boolean
}